Sources/OpenAIKit/Chat/Chat.swift

```diff
@@ -26,13 +26,19 @@ extension Chat.Choice: Codable {}
 
 extension Chat {
     public struct Message {
-        public let role: String
+        public let role: Role
         public let content: String
-        public init(role: String, content: String) {
+        public init(role: Role, content: String) {
             self.role = role
             self.content = content
         }
     }
 }
 
 extension Chat.Message: Codable {}
+
+extension Chat.Message {
+    public enum Role: String, Codable {
+        case system, user, assistant
```

Comment:
Put these on new lines :)

Reply:
Done. I'm thinking we could simplify it even more by making Message an enum, like so:
```swift
extension Chat {
    public enum Message: Codable {
        case system(String)
        case user(String)
        case assistant(String)
    }
}
```
What do you think?

Reply:
Yeah I like this change

Reply:
Cool. I'll do that

Reply:
@dylanshine I found a number of minor issues. I've updated more code than I anticipated this morning. Can you take a good look at the changes?

- Refactor Message and Finishreason to enums
- Make chat parameters except model and messages optional (so we don’t accidentally set both temperature and topP)
- Fix bug that didn't decode token usage correctly
- Fix date bug (correct format is secondsSince1970, not milliseconds)
- Fix key spelling error (stop instead of stops)
- Add unit test for JSON chat encoding/decoding

Reply:
@ronaldmannak As it relates to "Make chat parameters except model and messages optional (so we don’t accidentally set both temperature and topP)", I would really prefer to keep API non-optional with default values rather than provide optionals (unless the API behaves differently). For example, `stop` was implemented as `stops` with an empty `[String]` with custom encoding that would exclude it if it was empty. To handle "accidentally set both temperature and topP", I would prefer we use an enum to represent passing either value rather than two optional parameters that would lets say throw an assertion if both were set.

Reply:
Re: optionals, we can definitely pass the default values by default, but we'll have to keep track of the default values in the OpenAI documentation just in case OpenAI changes them (for whatever reason). I personally would prefer to send only the parameters that have non-default values, just as the examples in the OpenAI documentation. but can obviously revert back to the way it was if you want to.

I might have deleted the stops -> stop custom encoding. I deleted a few that weren't necessary. Less code = less bugs. Happy to revert this change as well if you want to.

Re: enum for topP and temperature, I thought about that but wasn't sure if perhaps someone would purposely set both to get a desired weird effect. But definitely could add an enum.

Reply:
"we'll have to keep track of the default values in the OpenAI documentation just in case OpenAI changes them" Is also a super fair point, let me think on it...

Is there a known valid use case for passing both topP and temperature?

Reply:
@ronaldmannak Let's keep the API consistent for now by reverting back to non-optional parameters where the API docs state a default value. For topP and temperature, let's let the API consumers experiment, and if we run into issues in the future we'll address them then. I very much appreciate the time and effort you've put into this PR

Reply:
@ronaldmannak Just following up on this, I can revert the changes if you're unavailable.

Reply:
@dylanshine Apologies. Got caught up in other things. Feel free to revert any changes.